src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -158,11 +160,12 @@ out instanceByteSizeAndAlignment
 
             // At this point all special cases are handled and all inputs validated
 
-            if (type.IsExplicitLayout)
+            if (!TargetCoreCLR && type.IsExplicitLayout)
             {
                 return ComputeExplicitFieldLayout(type, numInstanceFields);
             }
-            else if (type.IsSequentialLayout || type.Context.Target.Abi == TargetAbi.ProjectN || type.Context.Target.Abi == TargetAbi.CppCodegen)
+            else if (!TargetCoreCLR && (type.IsSequentialLayout || type.Context.Target.Abi == TargetAbi.ProjectN || type.Context.Target.Abi == TargetAbi.CppCodegen) ||
```

Comment:
This is getting _really_ hard to read.

Since you're adding a new virtual method anyway, I would suggest just placing everything after the `// At this point all special cases are handled and all inputs validated` into a virtual method and overriding it in the ready to run algorithm to do the weird thing (ignore what the user asked for). You'll need to make the methods that do sequential/auto/explicit layout `protected`.